src/tests/end2end/CopyTests.cpp

```diff
@@ -87,51 +88,63 @@ class CopyTests_T2B : public CopyTests {
+            std::vector<std::vector<RGBA8>> textureArrayData(textureSpec.arrayLayer);
+            std::vector<dawn::CommandBuffer> commands(textureSpec.arrayLayer * 2);
+            for (uint32_t slice = 0; slice < textureSpec.arrayLayer; ++slice) {
+                textureArrayData[slice].resize(texelCountPerLayer);
+                FillTextureData(width, height, rowPitch / kBytesPerTexel, textureArrayData[slice].data(), slice);
+                dawn::Buffer uploadBuffer = utils::CreateBufferFromData(device, textureArrayData[slice].data(),
+                    static_cast<uint32_t>(sizeof(RGBA8) * textureArrayData[slice].size()), dawn::BufferUsageBit::TransferSrc);
+
+                commands[slice] = device.CreateCommandBufferBuilder()
+                    .CopyBufferToTexture(uploadBuffer, 0, rowPitch, texture, 0, 0, 0, width, height, 1, textureSpec.level, slice)
+                    .GetResult();
+            }
 
-            // Create a buffer of size `size` and populate it with empty data (0,0,0,0)
+            // Create a buffer of size `size * textureSpec.arrayLayer` and populate it with empty data (0,0,0,0)
             // Note: Prepopulating the buffer with empty data ensures that there is not random data in the expectation
             // and helps ensure that the padding due to the row pitch is not modified by the copy
             dawn::BufferDescriptor bufDescriptor;
-            bufDescriptor.size = bufferSpec.size;
+            bufDescriptor.size = bufferSpec.size * textureSpec.arrayLayer;
             bufDescriptor.usage = dawn::BufferUsageBit::TransferSrc | dawn::BufferUsageBit::TransferDst;
             dawn::Buffer buffer = device.CreateBuffer(&bufDescriptor);
-
-            std::vector<RGBA8> emptyData(bufferSpec.size / kBytesPerTexel);
+            std::vector<RGBA8> emptyData(bufferSpec.size / kBytesPerTexel * textureSpec.arrayLayer);
             buffer.SetSubData(0, static_cast<uint32_t>(emptyData.size() * sizeof(RGBA8)), reinterpret_cast<const uint8_t*>(emptyData.data()));
 
-            // Copy the region [(`x`, `y`), (`x + copyWidth, `y + copyWidth`)] from the `level` mip into the buffer at the specified `offset` and `rowPitch`
-            commands[1] = device.CreateCommandBufferBuilder()
-                .CopyTextureToBuffer(texture, textureSpec.x, textureSpec.y, 0, textureSpec.copyWidth, textureSpec.copyHeight, 1, textureSpec.level, buffer, bufferSpec.offset, bufferSpec.rowPitch)
-                .GetResult();
+            uint32_t bufferOffset = bufferSpec.offset;
+            for (uint32_t slice = 0; slice < textureSpec.arrayLayer; ++slice) {
+                // Copy the region [(`x`, `y`), (`x + copyWidth, `y + copyWidth`)] from the `level` mip into the buffer at `offset + bufferSpec.size * slice` and `rowPitch`
+                commands[textureSpec.arrayLayer + slice] = device.CreateCommandBufferBuilder()
```

Comment:
this can still be in the same single command buffer as the uploads.

Reply:
Done